From d5f28720537d3dc0363d3da4320c42cf57d53925 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Tue, 7 Jan 2020 17:12:07 +0100
Subject: [PATCH] PONRTSYS-5630: Set bMacLearningLimitEnable when active

bMacLearningLimitEnable was not set when the limitation as activated
per port.
---
 drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
index 0adb929ba5fc..d5c1e2edf25e 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.c
@@ -23924,8 +23924,10 @@ GSW_return_t GSW_BridgePortConfigGet(void *cdev, GSW_BRIDGE_portConfig_t *param)
 	if (param->eMask & GSW_BRIDGE_PORT_CONFIG_MASK_BRIDGE_ID)
 		param->nBridgeId = ((tbl_prog_brdgeport_ingress.val[4] & 0x3F00) >> 8);
 
-	if (param->eMask & GSW_BRIDGE_PORT_CONFIG_MASK_MAC_LEARNING_LIMIT)
+	if (param->eMask & GSW_BRIDGE_PORT_CONFIG_MASK_MAC_LEARNING_LIMIT) {
 		param->nMacLearningLimit = (tbl_prog_brdgeport_ingress.val[4] & 0xFF);
+		param->bMacLearningLimitEnable = param->nMacLearningLimit != 255;
+	}
 
 	if (param->eMask & GSW_BRIDGE_PORT_CONFIG_MASK_INGRESS_VLAN) {
 		param->bIngressExtendedVlanEnable = ((tbl_prog_brdgeport_ingress.val[1] & 0x4000) >> 14);
